### DISCUSSION TITLE
Missing optional params props and meta in extended routes

### DISCUSSION THREAD

**Discussion by Lord-Y** [nuxt/nuxt#18119] [Questions]
Hi @danielroe following your PR https://github.com/nuxt/framework/pull/1740 , I'like to know it possible to add optional params `props` and `meta`  from `@nuxt/kit`  private repo?

Here are the actual params:
`declare type NuxtPage = {
    name?: string;
    path: string;
    file: string;
    children?: NuxtPage[];
};`

I actually want to port a a Vue 3 project to Nuxt 3 and these params are missing.
It would be great as `@nuxtjs/router` is incompatible with Nuxt 3.

Waiting for answer,
Thx.

**Comment by danielroe** [comment]
Could you clarify what your feature request is?

**Reply by Lord-Y** [reply]
@danielroe the idea is able to use extended routes like so:

```
import { defineNuxtModule, extendPages } from '@nuxt/kit'
import { resolve } from 'pathe'

export default defineNuxtModule({
  setup () {
    extendPages((pages) => {
      // Add /test page
      pages.push({
        name: 'Test',
        path: '/test',
        file: resolve(__dirname, './pages/test.vue'),
        meta: { requiresAuth: true },
        props: { default: true, sidebar: false }
      })
    })
  }
}
```

And the in my vue files to be able to use these datas.
It's the same idea as [route meta fields](https://next.router.vuejs.org/guide/advanced/meta.html#route-meta-fields) and [props in object mode](https://next.router.vuejs.org/guide/essentials/passing-props.html#object-mode)

**Reply by Lord-Y** [reply]
So @danielroe this feature is not possible right?

**Comment by jonahbohlmann** [comment]
My question is the same, I think. How can I write custom routes like in vue-router?
For example to map:
/my-page to /pages/theTechnicalName.vue? Any idea?
